### PR TITLE
Full button functionality

### PR DESCRIPTION
**Pull request by agreenspan**

### CODE REVIEW

File: Button.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        self.depressed = False
+        self.timer = timerThread
+        self.stop = threading.Event()
+
+    def run(self):
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setup(self.channel, GPIO.IN, pull_up_down=GPIO.PUD_UP)
+        GPIO.add_event_detect(self.channel, GPIO.FALLING, callback=self.edge_callback)
+
+        while True:
```
Comment (tahosa) [inline Button.py]: This doesn't have to be a loop. Once you set up the event trigger, you can just `self.stop.wait()`

File: Button.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            logging.info("Stopping thread")
+            break
+        return True
+
+    def edge_callback(self, channel):
+        if not GPIO.input(channel) and not self.depressed:
+            self.depressed = True
+            self.timer.pause.clear() if self.timer.pause.isSet() else self.timer.pause.set()
+            counter = 0
+            while not GPIO.input(channel):
```
Comment (tahosa) [inline Button.py]: Add some comments explaining what this does. Presumably, when the button is held for 8/10ths of a second it resets the timer, and when it's held for 2.5 seconds, it gets disabled.

File: Controller.py [hunk 3]
```diff
@@ -86,4 +86,4 @@ def run(self):
                 
             return True
                 
-        return True
+        return True
```
Comment (tahosa) [inline Controller.py]: Add the trailing newline

File: RoundTimer.py [hunk 4]
```diff
@@ -17,16 +18,19 @@
...
 
 def kill(signal, frame):
     '''
     Trap the signal and kill any ongoing threads, then exit
     '''
     for thread in threads:
+        try:
+            thread.pause.set()
+        except:
+            pass
```
Comment (tahosa) [inline RoundTimer.py]: Explain why we're not handling any errors in a comment.

File: RoundTimer.py [hunk 5]
```diff
@@ -42,21 +46,22 @@ def kill(signal, frame):
 # Tread creation
 logging.info("Creating threads")
 clkThread = Clock(clock, (64, 32), position=(0, 0), color=(255, 0, 0, 255))
-threads.append(clkThread)
 
-tmrThread1 = Timer(timers, (32, 16), minutes=1, position=(0, 0), offset=(0, -6), color=(0, 0, 255, 255))
-tmrThread2 = Timer(timers, (32, 16), minutes=1, position=(0, 16), offset=(0, -6), color=(255, 255, 0, 255))
-tmrThread3 = Timer(timers, (32, 16), minutes=2, position=(32, 0), offset=(0, -6), color=(255, 0, 0, 255))
-tmrThread4 = Timer(timers, (32, 16), minutes=2, position=(32, 16), offset=(0, -6), color=(255, 255, 255, 255))
+tmrThreads = []
```
Comment (tahosa) [inline RoundTimer.py:48]: At some point, we will want to move this to some sort of config file. For now, we can leave it as hard-coded but it needs to be more configurable going forward. Maybe open an issue once this merges?

File: RoundTimer.py [hunk 6]
```diff
@@ -42,21 +46,22 @@ def kill(signal, frame):
...
+tmrThreads.append(Timer(timers, (32, 16), minutes=50, position=(0, 0), offset=(0, -6), color=(0, 0, 255, 255)))
+tmrThreads.append(Timer(timers, (32, 16), minutes=50, position=(0, 16), offset=(0, -6), color=(255, 255, 0, 255)))
+tmrThreads.append(Timer(timers, (32, 16), minutes=50, position=(32, 0), offset=(0, -6), color=(255, 0, 0, 255)))
+tmrThreads.append(Timer(timers, (32, 16), minutes=50, position=(32, 16), offset=(0, -6), color=(255, 255, 255, 255)))
 
-threads.append(tmrThread1)
-threads.append(tmrThread2)
-threads.append(tmrThread3)
-threads.append(tmrThread4)
+btnThreads = []
```
Comment (tahosa) [inline RoundTimer.py:54]: Same as above about making this configurable.

File: Button.py (hunk 1 above)
Comment (agreenspan) [inline Button.py]: good call. it seemed like the thread was terminating without something keeping it open

File: RoundTimer.py (hunk 4 above)
Comment (agreenspan) [inline RoundTimer.py]: only some threads have pause we just want to try to do all of those and dont care about the others but i'll add comment

Comment (tahosa) [inline RoundTimer.py]: If only some thread have it, do something like

``` python
if thread.pause:
    thread.pause.set()
```

**Comment by agreenspan** [comment]
Woot

**Pull request merged**